rust/src/lib.rs

```diff
@@ -130,37 +125,39 @@ pub fn compare(left: &str, right: &str) -> Ordering {
             }
             (Some(l), Some(r)) => match (l, r) {
                 (Numerical(l, _), Numerical(r, _)) => l.cmp(&r),
-                (l, r) => l.to_string().cmp(&r.to_string()),
+                (l, r) => l.into_string().cmp(&r.into_string()),
             },
             (None, None) => unreachable!(),
         };
+
         if current != Equal {
             return current;
         }
     }
-    return Equal;
+
+    Equal
```

Comment:
This is the only change that I am apprehensive about in this entire thing.
To me the explicit `return` feels more readable, considering that everything else is in a decently nested block with itself uses `return`.